File: Models/Entities.cs
using LiteDB;

namespace FamilyPlanner.Models;

public abstract class EntityBase
{
    [BsonId]
    public int Id { get; set; }

    public int UserId { get; set; } = 1;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

public sealed class HouseholdProfile
{
    [BsonId]
    public int Id { get; set; } = 1;
    public string FamilyName { get; set; } = string.Empty;
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

public sealed class FamilyMember : EntityBase
{
    public string Name { get; set; } = string.Empty;
    public string Color { get; set; } = "#3b82f6";
    public string? AvatarUrl { get; set; }
    public string? Birthday { get; set; }
    public string? Bio { get; set; }
}

public sealed class PlannerEvent : EntityBase
{
    public string Title { get; set; } = string.Empty;
    public string EventDate { get; set; } = string.Empty;
    public string? StartTime { get; set; }
    public string? EndTime { get; set; }
    public int? OwnerId { get; set; }
    public string? Color { get; set; }
    public string? Note { get; set; }
}

public sealed class MealPlan : EntityBase
{
    public int DayOfWeek { get; set; }
    public string MealType { get; set; } = "dinner";
    public string Meal { get; set; } = string.Empty;
    public int? OwnerId { get; set; }
    public string? Note { get; set; }
}

public sealed class BudgetMonth : EntityBase
{
    public string Month { get; set; } = string.Empty;
    public decimal Limit { get; set; }
    public decimal Income { get; set; }
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

public sealed class ExpenseItem : EntityBase
{
    public decimal Amount { get; set; }
    public string Category { get; set; } = "Annet";
    public string? ExpenseDate { get; set; }
    public int? OwnerId { get; set; }
    public string? Description { get; set; }
    public string Month { get; set; } = string.Empty;
}

public sealed class MedicineItem : EntityBase
{
    public string Name { get; set; } = string.Empty;
    public string? Time { get; set; }
    public int? OwnerId { get; set; }
    public string? Note { get; set; }
    public bool Taken { get; set; }
}

public sealed class NoteItem : EntityBase
{
    public string Title { get; set; } = string.Empty;
    public int? OwnerId { get; set; }
    public string? Content { get; set; }
}

public sealed class ShoppingItem : EntityBase
{
    public string Item { get; set; } = string.Empty;
    public int? OwnerId { get; set; }
    public int Quantity { get; set; } = 1;
    public bool Done { get; set; }
    public int? SourceMealId { get; set; }
}

public sealed class FamilyAssignment : EntityBase
{
    public int DayOfWeek { get; set; }
    public int FamilyMemberId { get; set; }
    public string ActivityType { get; set; } = "medicine";
    public string? Note { get; set; }
}

public sealed class BudgetSnapshot
{
    public decimal Limit { get; set; }
    public decimal Income { get; set; }
    public decimal Spent { get; set; }
    public decimal Remaining { get; set; }
    public decimal Available { get; set; }
    public IReadOnlyList<ExpenseItem> Expenses { get; set; } = [];
}

public sealed class AssignmentEnvelope
{
    public IReadOnlyList<FamilyAssignment> Assignments { get; set; } = [];
}

File: android/app/src/main/kotlin/io/github/fourgnusd3i/familyplanner/data/local/Entities.kt
package io.github.fourgnusd3i.familyplanner.data.local

import androidx.room.Entity
import androidx.room.Index
import androidx.room.PrimaryKey
import java.math.BigDecimal
import java.time.Instant
import java.time.LocalDate
import java.time.LocalTime

@Entity(tableName = "household_profiles")
data class HouseholdProfileEntity(
    @PrimaryKey val id: Long = 1,
    val familyName: String,
    val createdAt: Instant = Instant.now(),
)

@Entity(tableName = "family_members")
data class FamilyMemberEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val name: String,
    val color: String = "#3b82f6",
    val avatarUri: String? = null,
    val birthday: LocalDate? = null,
    val bio: String? = null,
    val createdAt: Instant = Instant.now(),
)

@Entity(
    tableName = "planner_events",
    indices = [
        Index("eventDate"),
        Index("ownerId"),
        Index("recurrenceType"),
        Index("sourceType"),
        Index("sourceMemberId"),
        Index("sourceYear"),
    ],
)
data class PlannerEventEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val eventDate: LocalDate,
    val startTime: LocalTime? = null,
    val endTime: LocalTime? = null,
    val recurrenceType: String? = null,
    val recurrenceUntil: LocalDate? = null,
    val ownerId: Long? = null,
    val color: String? = null,
    val note: String? = null,
    val sourceType: String? = null,
    val sourceMemberId: Long? = null,
    val sourceYear: Int? = null,
    val seriesStartDate: LocalDate? = null,
    val createdAt: Instant = Instant.now(),
)

@Entity(tableName = "meal_plans", indices = [Index("dayOfWeek"), Index("ownerId")])
data class MealPlanEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val dayOfWeek: Int,
    val mealType: String,
    val meal: String,
    val ownerId: Long? = null,
    val note: String? = null,
    val createdAt: Instant = Instant.now(),
)

@Entity(tableName = "budget_months", indices = [Index(value = ["month"], unique = true)])
data class BudgetMonthEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val month: String,
    val limit: BigDecimal,
    val income: BigDecimal,
    val currencyCode: String,
    val updatedAt: Instant = Instant.now(),
    val createdAt: Instant = Instant.now(),
)

@Entity(tableName = "expenses", indices = [Index("month"), Index("ownerId")])
data class ExpenseEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val amount: BigDecimal,
    val category: String,
    val expenseDate: LocalDate,
    val ownerId: Long? = null,
    val description: String? = null,
    val month: String,
    val createdAt: Instant = Instant.now(),
)

@Entity(tableName = "notes", indices = [Index("ownerId"), Index("createdAt")])
data class NoteEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val title: String,
    val ownerId: Long? = null,
    val content: String? = null,
    val createdAt: Instant = Instant.now(),
)

@Entity(tableName = "shopping_items", indices = [Index("ownerId"), Index("done"), Index("doneAt")])
data class ShoppingItemEntity(
    @PrimaryKey(autoGenerate = true) val id: Long = 0,
    val item: String,
    val ownerId: Long? = null,
    val quantity: Int = 1,
    val done: Boolean = false,
    val doneAt: Instant? = null,
    val createdAt: Instant = Instant.now(),
)
